### PR TITLE
fix: reproducible build (#1066, #1089)

### PR DESCRIPTION
**Pull request by Jack-Works**

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -73,7 +73,18 @@ module.exports = (argvEnv, argv) => {
     const dist = env === 'production' ? src('./build') : src('./dist')
 
     const gitInfo = () => {
-        if (target.ReproducibleBuild) return undefined
```
Comment (septs) [inline webpack.config.js:76]: plz use `@nice-labs/git-rev@3.4.2` new api
```typescript
isRepository(): boolean
```

Comment (septs) [reply webpack.config.js:76]:
```typescript
const environment = { ... }
if (!target.ReproducibleBuild && git.isRepository()) {
    Object.assign(environment, { ... })
}
return new webpack.EnvironmentPlugin(environment)
```

File: src/plugins/Wallet/real.ts [hunk 2]
```diff
@@ -460,6 +487,14 @@ export const walletAPI = {
                         txHash = hash
                     },
                     onReceipt() {
+                        if (receipt) {
```
Comment (septs) [inline src/plugins/Wallet/real.ts]:
```suggestion
                        if (!receipt) return
```

Comment (Jack-Works) [reply src/plugins/Wallet/real.ts]: @clearloop

Comment (clearloop) [reply src/plugins/Wallet/real.ts]: y

**Pull request merged**